### ISSUE TITLE
Expand docs on interaction of server side sessions between BFF and IdentityServer

### ISSUE THREAD

**Issue by josephdecock** [DuendeSoftware/docs.duendesoftware.com#329]
Turn this into docs: https://github.com/DuendeSoftware/Support/issues/778#issuecomment-1650306118

**Comment by khalidabuhakmeh** [comment]
@josephdecock would you like me to add this to the documentation, and what parts of the conversation do you feel are most important to add?

**Comment by josephdecock** [comment]
That would be great! 

The thrust of the idea is getting people to understand that, while "me sitting at my computer" is 1 logical session, there are n+1 physical sessions required - each client application has its own session state, and so does IdentityServer. Each of those pieces of session data can be made into a "server-side session", independently.

Server side sessions at the clients allow the client to
- receive back channel logout notifications
- forcibly end a user's session of that app (but note that the IdentityServer session might get them signed back in again, immediately)

Server side sessions at IdentityServer allow for a lot of powerful things
- Receive back channel logout notifications from upstream idps in a federation
- forcibly end a user's session at IdentityServer (but not that the client app might still have their own session)
- Global inactivity timeout across SSO apps (so basically, no client has interacted with identity server for too long, so log them all out).

The essence of it is that, sometimes you need a server side session at only IdentityServer, sometimes only at the client apps, and sometimes both. 

There's a lot of "it depends here", but I think people tend to be confused about the fundamentals - both conceptually (that there are distinct physical sessions, and what they're for) and concretely (how ASP.NET handles this).

I think the comment within the thread that I linked to, and especially this part, are the most important:

> > I'm still a little confused about part of this response. You talk about choosing whether to do server side sessions in each client, however the docs and samples present server sidce sessions as a single configuration for the entire identity server

> 
> The choice of whether to use server side sessions at the clients is made within each client application, not within the client configuration of IdentityServer. If your client applications are ASP.NET applications, then they're using its cookie authentication handler to store their authentication tickets - session data that is specific to each client application. The authentication ticket includes an identity made up of claims (which each client application independently can customize through claims transformation, etc) and metadata, such as when the user logged in, and their tokens (which are also specific to each application). Some of this data will be similar if a user uses single sign in to go to another client application, but fundamentally each application is managing its own session state.
> 
> That session state could be stored entirely within the cookie, which is the default, or you could implement the ITicketStore to store the session data on the server side (in redis, etc). Microsoft has an example that stores the data in memory (obviously not suitable for production) [here](https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Security/Authentication/Cookies/samples/CookieSessionSample/MemoryCacheTicketStore.cs). But each application has its own authentication ticket, and its own ticket store.
> 
> Similarly, your IdentityServer implementation is a .NET application using the cookie handler to track its own session. The Server-Side Session feature in IdentityServer is basically an implementation of the ITicketStore for use within IdentityServer itself, along with integration and services within IdentityServer that make use of the tickets. This is an implementation of the ITicketStore just for IdentityServer. Client applications that want an ITicketStore have to build that themselves (or you might have clients built with any other technology stack, and the implementation details of those client's sessions will vary).

**Comment by khalidabuhakmeh** [comment]
Do you think the expansion would go here? 

https://github.com/DuendeSoftware/docs.duendesoftware.com/blob/aa9e760592a5651c6a897a598e199da77b15325e/IdentityServer/v7/docs/content/bff/overview.md

**Comment by josephdecock** [comment]
Yes, and maybe cross reference (and possible update) the identity server server side session pages.

**Comment by maartenba** [comment]
Fixed in [#574](https://github.com/DuendeSoftware/docs.duendesoftware.com/pull/574)
